transport/src/test/java/io/camunda/zeebe/transport/stream/impl/ClientStreamRequestManagerTest.java

```diff
@@ -0,0 +1,137 @@
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import io.atomix.cluster.MemberId;
+import io.atomix.cluster.messaging.ClusterCommunicationService;
+import io.camunda.zeebe.scheduler.testing.TestConcurrencyControl;
+import io.camunda.zeebe.transport.stream.impl.messages.StreamTopics;
+import io.camunda.zeebe.util.buffer.BufferReader;
+import io.camunda.zeebe.util.buffer.BufferUtil;
+import io.camunda.zeebe.util.buffer.BufferWriter;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import org.agrona.MutableDirectBuffer;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class ClientStreamRequestManagerTest {
+
+  private final ClusterCommunicationService mockTransport = mock(ClusterCommunicationService.class);
+  private final ClientStreamRequestManager<BufferWriter, BufferReader> requestManager =
+      new ClientStreamRequestManager<>(mockTransport, new TestConcurrencyControl());
+
+  private final ClientStream<BufferWriter, BufferReader> clientStream =
+      new ClientStream<>(
+          UUID.randomUUID(), BufferUtil.wrapString("foo"), new TestMetadata(), p -> {});
+
+  @BeforeEach
+  void setup() {
+    when(mockTransport.send(any(), any(), any(), any(), any(), any()))
+        .thenReturn(CompletableFuture.completedFuture(null));
+  }
+
+  @Test
+  void shouldSendAddRequestToAllServers() {
+    // given
+    final MemberId server1 = MemberId.from("1");
+    final MemberId server2 = MemberId.from("2");
+    final var servers = Set.of(server1, server2);
+    // when
+    requestManager.openStream(clientStream, servers);
+
+    // then
+    verify(mockTransport)
+        .send(eq(StreamTopics.ADD.topic()), any(), any(), any(), eq(server1), any());
+    verify(mockTransport)
+        .send(eq(StreamTopics.ADD.topic()), any(), any(), any(), eq(server2), any());
+
+    assertThat(clientStream.isConnected(server1)).isTrue();
+    assertThat(clientStream.isConnected(server2)).isTrue();
+  }
+
+  @Test
+  void shouldRetryWhenAddRequestFails() {
+    // given
+    final MemberId server = MemberId.from("1");
+    final var servers = Set.of(server);
+    when(mockTransport.send(any(), any(), any(), any(), any(), any()))
+        .thenReturn(CompletableFuture.failedFuture(new RuntimeException("Expected")))
+        .thenReturn(CompletableFuture.completedFuture(null));
+
+    // when
+    requestManager.openStream(clientStream, servers);
+
+    // then
+    verify(mockTransport, times(2))
+        .send(eq(StreamTopics.ADD.topic()), any(), any(), any(), eq(server), any());
+    assertThat(clientStream.isConnected(server)).isTrue();
+  }
+
+  @Test
+  void shouldSendRemoveRequestToAllServers() {
+    // given
+    final MemberId server1 = MemberId.from("1");
+    final MemberId server2 = MemberId.from("2");
+    final var servers = Set.of(server1, server2);
+    requestManager.openStream(clientStream, servers);
```

Comment:
:upside_down_face: I would set up the test fixtures directly in the registry, especially when using mocks to avoid triggering unexpected code paths or messing with verification. Very soft suggestion because we only mock a single method :smile: